mmv1/api/resource.go

```diff
@@ -730,14 +730,90 @@ func (r Resource) GetIdentity() []*Type {
 	})
 }
 
-func (r *Resource) AddLabelsRelatedFields(props []*Type, parent *Type) []*Type {
+func buildWriteOnlyField(name string, originalField *Type, originalFieldLineage string) *Type {
+	description := fmt.Sprintf("%s Note: This property is write-only and will not be read from the API. For more info see [updating write-only attributes](/docs/providers/google/guides/using_write_only_attributes.html#updating-write-only-attributes)", google.Underscore(originalField.Description))
+	fieldPathOriginalField := originalFieldLineage
+	fieldPathCurrentField := strings.ReplaceAll(originalFieldLineage, google.Underscore(originalField.Name), google.Underscore(name))
+
+	apiName := originalField.ApiName
+	if apiName == "" {
+		apiName = originalField.Name
+	}
+
+	options := []func(*Type){
```

Comment:
It looks like we need to specify RequiredWith for the write only field, too - without it, I'm able to specify the write only field without the version field, which results in unexpected behavior (permadiff where the original field has a value set and Terraform keeps trying to unset it)

I guess this works opposite of how I expected it!